pkg/middleware/auth.go

```diff
@@ -47,7 +47,8 @@ func notAuthorized(c *m.ReqContext) {
 		return
 	}
 
-	c.SetCookie("redirect_to", url.QueryEscape(setting.AppSubUrl+c.Req.RequestURI), 0, setting.AppSubUrl+"/", nil, false, true)
+	c.Resp.Header().Del("Set-Cookie")
+	WriteCookie(c.Resp, "redirect_to", url.QueryEscape(setting.AppSubUrl+c.Req.RequestURI), 0, nil)
```

Comment:
This must include proper config as last argument. Know it's tricky since you only have `setting.Cfg` available in some handlers and some not. Maybe you can add override functions in middleware/cookie.go to either use `setting.Cfg` or global `setting` that we eventually want to move away from.